File: solutions/csharp/matrix/1/Matrix.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Matrix
{
    private readonly int[][] matrix;

    public Matrix(string input) => matrix = ParseMatrix(input);

    private int[][] ParseMatrix(string input)
    {
        return input
            .Split('\n', StringSplitOptions.RemoveEmptyEntries)
            .Select(row => row.Trim().Split(' ', StringSplitOptions.RemoveEmptyEntries)
                .Select(int.Parse)
                .ToArray())
            .ToArray();
    }

    public int[] Row(int row) => matrix[row - 1];

    public int[] Column(int col) => Enumerable.Range(0, matrix.Length).Select(row => matrix[row][col - 1]).ToArray();
}

File: solutions/python/matrix/1/matrix.py
class Matrix:
    def __init__(self, matrix_string):
        self.string_representation = matrix_string

    def row(self, index):
        string_rows = self.string_representation.split('\n')
        rows = []
        for row in string_rows:
            string_row = row.split()
            int_row = []
            for item in string_row:
                int_row.append(int(item))
            rows.append(int_row)
        return rows[index - 1]

    def column(self, index):
        string_rows = self.string_representation.split('\n')
        rows = []
        for row in string_rows:
            string_row = row.split()
            int_row = []
            for item in string_row:
                int_row.append(int(item))
            rows.append(int_row)
        column = []
        for row in rows:
            column.append(row[index - 1])
        return column
